src/Surfnet/StepupMiddleware/MiddlewareBundle/Console/Command/BootstrapIdentityWithSmsSecondFactorCommand.php

```diff
@@ -0,0 +1,220 @@
+            ->addArgument(
+                'phone-number',
+                InputArgument::REQUIRED,
+                'The phone number of the user should be formatted like "+31 (0) 612345678"'
+            )
+            ->addArgument(
+                'registration-status',
+                InputArgument::REQUIRED,
+                'Valid arguments: unverified, verified, vetted'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var Container $container */
+        $container = $this->getApplication()->getKernel()->getContainer();
+        $tokenStorage = $container->get('security.token_storage');
+        $identityRepository = $container->get('surfnet_stepup_middleware_api.repository.identity');
+        $unverifiedSecondFactorRepository = $container->get(
+            'surfnet_stepup_middleware_api.repository.unverified_second_factor'
+        );
+        $verifiedSecondFactorRepository = $container->get(
+            'surfnet_stepup_middleware_api.repository.verified_second_factor'
+        );
+        $pipeline = $container->get('surfnet_stepup_middleware_command_handling.pipeline.transaction_aware_pipeline');
+        $eventBus = $container->get('surfnet_stepup_middleware_command_handling.event_bus.buffered');
+        $connection = $container->get('surfnet_stepup_middleware_middleware.dbal_connection_helper');
+
+        $tokenStorage->setToken(
+            new AnonymousToken('cli.bootstrap-identity-with-sms-token', 'cli', ['ROLE_SS', 'ROLE_RA'])
+        );
+
+        $nameId = new NameId($input->getArgument('name-id'));
+        $institution = new Institution($input->getArgument('institution'));
+        $commonName = $input->getArgument('common-name');
+        $email = $input->getArgument('email');
+        $preferredLocale = $input->getArgument('preferred-locale');
+        $registrationStatus = $input->getArgument('registration-status');
+        $phoneNumber = $input->getArgument('phone-number');
+        $identity = false;
+
+        $output->writeln(
+            sprintf(
+                '<notice>Adding a %s SMS token for %s</notice>',
+                $registrationStatus,
+                $commonName
+            )
+        );
+
+        if ($identityRepository->hasIdentityWithNameIdAndInstitution($nameId, $institution)) {
+            $output->writeln(
+                sprintf(
+                    '<notice>An identity with name ID "%s" from institution "%s" already exists, using that identity</notice>',
+                    $nameId->getNameId(),
+                    $institution->getInstitution()
+                )
+            );
+            $identity = $identityRepository->findOneByNameIdAndInstitution($nameId, $institution);
+        }
+
+        $connection->beginTransaction();
+
+        $secondFactorId = Uuid::uuid4()->toString();
+
+        if (!$identity) {
```

Comment:
I think it will be better to remove this, because you should bootstrap an identity and not update an identity so it will be better to throw an exception.

Reply:
You may be right, but at the other hand, it would be rather tedious to script up a system to first create/activate a user, and then register a token. 

I'm asking PMs what their preference is. From a single responsibility standpoint I fully agree with you by the way.

Reply:
I don't mean to do either one of them (create identity or vet a token) but always do both, so please remove this check and always create an identity with a token. So if the Identity already exists throw an exception.

Reply:
The benefit of this system is that you can also create multi-token users this way. Something that is interesting from a test perspective.

Reply:
@pmeulen and/or @phavekes what's your preference?

Reply:
Discussed this. We decided upon a solution where we have separate commands for identity and token cration.